src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -322,6 +322,17 @@ public function stream($responses, float $timeout = null): ResponseStreamInterfa
             throw new \TypeError(sprintf('%s() expects parameter 1 to be an iterable of CurlResponse objects, %s given.', __METHOD__, \is_object($responses) ? \get_class($responses) : \gettype($responses)));
         }
 
+        foreach ($this->multi->openHandles as [$ch, $options]){
```

Comment:
Really not sure about that. I will cripple the perf of the client when hundred of responses are streamed together.

Reply:
refactored:
- added a property in ClientState to track handle with resources only
- use an error_handler to performs the check on failure only